integrationTests/chainSimulator/staking/stakingProvider/delegationMigrate_test.go

```diff
@@ -0,0 +1,127 @@
+	roundDurationInMillis := uint64(6000)
+	roundsPerEpoch := core.OptionalUint64{
+		HasValue: true,
+		Value:    20,
+	}
+
+	cs, err := chainSimulator.NewChainSimulator(chainSimulator.ArgsChainSimulator{
+		BypassTxSignatureCheck: true,
+		TempDir:                t.TempDir(),
+		PathToInitialConfig:    defaultPathToInitialConfig,
+		NumOfShards:            3,
+		GenesisTimestamp:       time.Now().Unix(),
+		RoundDurationInMillis:  roundDurationInMillis,
+		RoundsPerEpoch:         roundsPerEpoch,
+		ApiInterface:           api.NewNoApiInterface(),
+		MinNodesPerShard:       1,
+		MetaChainMinNodes:      1,
+		AlterConfigsFunction: func(cfg *config.Configs) {
+			configs.SetStakingV4ActivationEpochs(cfg, 2)
+		},
+	})
+	require.Nil(t, err)
+	require.NotNil(t, cs)
+	defer cs.Close()
+
+	mintValue := big.NewInt(0).Mul(big.NewInt(5000), chainSimulatorIntegrationTests.OneEGLD)
+	sp1Owner, err := cs.GenerateAndMintWalletAddress(0, mintValue)
+	require.Nil(t, err)
+
+	sp2Owner, err := cs.GenerateAndMintWalletAddress(0, mintValue)
+	require.Nil(t, err)
+
+	delegator, err := cs.GenerateAndMintWalletAddress(0, mintValue)
+	require.Nil(t, err)
+
+	err = cs.GenerateBlocks(1)
+	require.Nil(t, err)
+
+	err = cs.GenerateBlocksUntilEpochIsReached(1)
+	require.Nil(t, err)
+
+	// create sp1
+	stakeValue, _ := big.NewInt(0).SetString("4250000000000000000000", 10)
+	dataField := "createNewDelegationContract@00@0ea1"
+	txStake := chainSimulatorIntegrationTests.GenerateTransaction(sp1Owner.Bytes, staking.GetNonce(t, cs, sp1Owner), vm.DelegationManagerSCAddress, stakeValue, dataField, 80_000_000)
+	stakeTx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(txStake, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, stakeTx)
+
+	delegationAddress := stakeTx.Logs.Events[2].Address
+	sp1AddressBytes, _ := cs.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(delegationAddress)
+
+	txStake = chainSimulatorIntegrationTests.GenerateTransaction(sp2Owner.Bytes, staking.GetNonce(t, cs, sp2Owner), vm.DelegationManagerSCAddress, stakeValue, dataField, 80_000_000)
+	stakeTx, err = cs.SendTxAndGenerateBlockTilTxIsExecuted(txStake, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, stakeTx)
+
+	delegationAddress = stakeTx.Logs.Events[2].Address
+	sp2AddressBytes, _ := cs.GetNodeHandler(0).GetCoreComponents().AddressPubKeyConverter().Decode(delegationAddress)
+
+	fmt.Println(sp1AddressBytes, sp2AddressBytes)
+
+	// do delegate to sp1
+	delegateValue := big.NewInt(0).Mul(big.NewInt(500), chainSimulatorIntegrationTests.OneEGLD)
+	txDelegate1 := chainSimulatorIntegrationTests.GenerateTransaction(delegator.Bytes, 0, sp1AddressBytes, delegateValue, "delegate", gasLimitForDelegate)
+	delegate1Tx, err := cs.SendTxAndGenerateBlockTilTxIsExecuted(txDelegate1, staking.MaxNumOfBlockToGenerateWhenExecutingTx)
+	require.Nil(t, err)
+	require.NotNil(t, delegate1Tx)
+
+	err = cs.ChangeEpochs(40)
```

Comment:
I would add a failing test here, to see that moving in the 30 epochs after the initial stake is also not possible.

Reply:
this check is done at line 104

Reply:
ok... so you can move instantly after the first stake. No migrations are allowed with less than 30 epochs